### ISSUE TITLE
`from module import submodule` implicit reexport is allowed

### ISSUE THREAD

**Issue by mikeroll** [python/mypy#12689]
**Bug Report**
`mypy --no-implicit-reexport` doesn't catch reexports written in the form of `from module import submodule`.

**To Reproduce**

```python
# pkg/lib.py

# implicit re-exports
import requests
import requests as demands
import requests.status_codes as requests_status_codes
from requests import Request

# implicit submodule re-export, but mypy doesn't care
from requests import status_codes
```

```python
# pkg/shell.py

from pkg.lib import requests, demands, requests_status_codes, Request, status_codes
```

**Expected Behavior**
I would expect each of those reexports to trigger an error.

**Actual Behavior**
`from requests import status_codes` is ignored.

```bash
❯ mypy --no-implicit-reexport pkg
pkg/shell.py:1: error: Module "pkg.lib" does not explicitly export attribute "requests"; implicit reexport disabled
pkg/shell.py:1: error: Module "pkg.lib" does not explicitly export attribute "demands"; implicit reexport disabled
pkg/shell.py:1: error: Module "pkg.lib" does not explicitly export attribute "requests_status_codes"; implicit reexport disabled
pkg/shell.py:1: error: Module "pkg.lib" does not explicitly export attribute "Request"; implicit reexport disabled
Found 4 errors in 1 file (checked 3 source files)
```

**Your Environment**

- Mypy version used: 0.942
- Mypy command-line flags: `--no-implicit-reexports`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.9.10
- Operating system and version: Arch Linux latest

**Comment by hauntsaninja** [comment]
Thanks for the report! It's quite surprising to me, but looks like this has always been mypy's behaviour. I opened a potential fix here #12704, so we'll find out if I'm missing something
